subxt/src/client/online_client.rs

```diff
@@ -437,17 +438,13 @@ impl<T: Config> RuntimeUpdaterStream<T> {
             Err(err) => return Some(Err(err)),
         };
 
-        let latest_block_ref = match self.client.backend().latest_finalized_block_ref().await {
-            Ok(block_ref) => block_ref,
-            Err(e) => return Some(Err(e)),
+        let at = match wait_for_runtime_upgrade(&self.client, &runtime_version).await {
```

Comment:
Nit; maybe rename to `wait_for_finalized_block_containing_update` or something like that